src/client/components/home/Home.js

```diff
@@ -1,27 +1,120 @@
+import InputBase from '@material-ui/core/InputBase';
 import Typography from '@material-ui/core/Typography';
-import React, { useContext } from 'react';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { APP_AUTHOR, APP_DESCRIPTION, APP_NAME } from '../../config/constants';
 import CollectionsGrid from '../collection/CollectionsGrid';
-import { CollectionContext } from '../CollectionProvider';
 import Seo from '../common/Seo';
+import { useCollections } from '../../utils/collections';
+import Loader from '../common/Loader';
 
 const useStyles = makeStyles(() => ({
   wrapper: {
     padding: '4vw',
   },
+  root: {
+    padding: '2px 4px',
+    margin: '100px 0px',
+    display: 'flex',
+    alignItems: 'center',
+    width: '100%',
+  },
+  input: {
+    marginLeft: '20px',
+    flex: 6,
+  },
+  iconButton: {
+    padding: 10,
+  },
+  divider: {
+    height: 28,
+    margin: 4,
+  },
+  typographyMargin: {
+    margin: '10px 0px',
+  },
 }));
 
 function Home() {
   const { t } = useTranslation();
-  const { collections } = useContext(CollectionContext);
+  const [queried, setQueried] = useState(false);
+  const [searchResults, setSearchResults] = useState(null);
+  const { collections, isLoading } = useCollections();
+
+  const handleSearchClick = (event) => {
+    event.preventDefault();
+    const query = event.target.elements.search.value.toLowerCase();
+    if (query.length > 0) {
+      setSearchResults(
+        collections.filter(
+          (collection) =>
+            collection.name.toLowerCase().includes(query) ||
+            collection.author?.name.toLowerCase().includes(query),
+        ),
+      );
+      setQueried(true);
+    }
+  };
   const classes = useStyles();
   return (
     <>
       <Seo title={APP_NAME} description={APP_DESCRIPTION} author={APP_AUTHOR} />
       <div className={classes.wrapper}>
-        <Typography variant="h2">{t('Home')}</Typography>
+        <Typography variant="h3" align="center">
+          {t('Graasp Collections Directory')}
+        </Typography>
+        <Paper
+          component="form"
+          className={classes.root}
+          onSubmit={handleSearchClick}
+        >
+          <InputBase
+            id="search"
+            disabled={isLoading}
+            style={{ margin: 8 }}
+            placeholder="Search collections using name or author"
+            fullWidth
+            margin="normal"
+            InputLabelProps={{
+              shrink: true,
+              ariaLabel: 'search collections',
+            }}
+            variant="filled"
+          />
+          <Divider className={classes.divider} orientation="vertical" />
+          <IconButton
+            color="primary"
+            className={classes.iconButton}
+            aria-label="directions"
+            type="submit"
+          >
+            <SearchIcon />
+          </IconButton>
+        </Paper>
+        {isLoading ? (
```

Comment:
Maybe include the results in `Paper`? It would avoid the massive margin between the search bar and the results vs `Discover` and `Results`.